scripts/fix-link-paymentIntents.js

```diff
@@ -0,0 +1,173 @@
+    return;
+  }
+
+  const slug = charge.metadata.to.replace('https://opencollective.com/', '');
+  const collective = await models.Collective.findOne({
+    where: { slug: slug },
+  });
+  if (!collective) {
+    console.log(`Ignoring ${charge.id} (could not find Collective ${slug})`);
+    return;
+  }
+
+  const transaction = await models.Transaction.findOne({
+    where: { CollectiveId: collective.id, data: { charge: { id: charge.id } } },
+    order: [['id', 'DESC']],
+  });
+
+  if (!transaction) {
+    console.error(`🚨️ Missing transaction for stripe charge ${charge.id}`);
+  }
+}
+*/
+
+const getHostStripeAccountUsername = async slug => {
+  const hostId = (await models.Collective.findOne({ where: { slug } }))?.id;
+  if (!hostId) {
+    throw new Error('Host not found');
+  }
+
+  const stripeAccount = await models.ConnectedAccount.findOne({ where: { service: 'stripe', CollectiveId: hostId } });
+  if (!stripeAccount) {
+    throw new Error('No stripe account found for this host');
+  }
+
+  return stripeAccount.username;
+};
+
+async function recordCharge(charge, paymentIntent, stripeAccount) {
+  // Copy code from Stripe webhook (paymentIntentSucceeded)
+  const order = await models.Order.findOne({
+    where: {
+      data: { paymentIntent: { id: paymentIntent.id } },
+    },
+    include: [
+      { association: 'collective', required: true },
+      { association: 'fromCollective', required: true },
+      { association: 'createdByUser', required: true },
+    ],
+  });
+
+  if (!order) {
+    logger.debug(`Stripe Webhook: Could not find Order for Payment Intent ${paymentIntent.id}`);
+    return;
+  }
+
+  // If charge was already processed, ignore event. (Potential edge-case: if the webhook is called while processing a 3DS validation)
+  const existingChargeTransaction = await models.Transaction.findOne({
+    where: { OrderId: order.id, data: { charge: { id: charge.id } } },
+  });
+  if (existingChargeTransaction) {
+    logger.info(`Stripe Webhook: ${order.id} already processed charge ${charge.id}, ignoring event ${event.id}`);
+    return;
+  }
+
+  await createOrUpdateOrderStripePaymentMethod(order, stripeAccount, paymentIntent);
+
+  const transaction = await createChargeTransactions(charge, { order });
+
+  // after successful first payment of a recurring subscription where the payment confirmation is async
+  // and the subscription is managed by us.
+  if (order.interval && !order.SubscriptionId) {
+    await createSubscription(order);
+  }
+
+  await order.update({
+    status: !order.SubscriptionId ? OrderStatuses.PAID : OrderStatuses.ACTIVE,
+    processedAt: new Date(),
+    data: {
+      ...omit(order.data, 'paymentIntent'),
+      previousPaymentIntents: [...(order.data.previousPaymentIntents ?? []), paymentIntent],
+    },
+  });
+
+  if (order.fromCollective?.ParentCollectiveId !== order.collective.id) {
+    await order.getOrCreateMembers();
+  }
+
+  sendEmailNotifications(order, transaction);
```

Comment:
This is the right thing to do, but maybe we should inform support about what's happening; some will likely be surprised and reach out.